src/lib/common/sem.cpp

```diff
@@ -83,14 +85,39 @@ int semInit(int shared, int takenInitially)
 *
 * reqSemTake -
 */
-int reqSemTake(const char* who, const char* what)
+int reqSemTake(const char* who, const char* what, SemRequestType reqType, bool* taken)
 {
   int r;
 
+  if (reqType == SemNoneOp)
```

Comment:
I didn't mean to ...
I guess I was thinking about reqPolicy and treated both of them the same way.
I will remove this part, thanks for discovering it :-)